### PR TITLE
*: add bottommost level compaction

### PR DESCRIPTION
**Pull request by huachaohuang**
Expose the bottommost level compaction option to make it more flexible.

### CODE REVIEW

File: librocksdb_sys/rocksdb/db/compaction_picker.cc [hunk 1]
```diff
@@ -677,7 +677,7 @@ Compaction* CompactionPicker::CompactRange(
       mutable_cf_options.max_compaction_bytes, output_path_id,
       GetCompressionType(ioptions_, vstorage, mutable_cf_options, output_level,
                          vstorage->base_level()),
-      /* max_subcompactions */ 0, std::move(grandparents),
+      max_subcompactions, std::move(grandparents),
```
Comment (zhangjinpeng1987) [inline librocksdb_sys/rocksdb/db/compaction_picker.cc]: Does RocksDB master branch has this feature?

File: librocksdb_sys/src/lib.rs [hunk 2]
```diff
@@ -292,6 +292,14 @@ pub enum DBTableProperty {
     CompressionName = 17,
 }
 
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+#[repr(C)]
+pub enum DBBottommostLevelCompaction {
+    Skip = 0,
```
Comment (zhangjinpeng1987) [inline librocksdb_sys/src/lib.rs]: Please give more explanations for these strategies.

File: librocksdb_sys/rocksdb/db/compaction_picker.cc (hunk 1 above)
Comment (huachaohuang) [reply librocksdb_sys/rocksdb/db/compaction_picker.cc]: https://github.com/facebook/rocksdb/pull/4082

Comment (huachaohuang) [reply librocksdb_sys/rocksdb/db/compaction_picker.cc]: This is fixed in RocksDB, PTAL.

**Comment by huachaohuang** [comment]
PTAL @zhangjinpeng1987 @DorianZheng

**Pull request merged**